main.go

```diff
@@ -100,8 +98,17 @@ func run(ctx context.Context, env Environment) error {
 		_ = database.Close()
 	}()
 
-	// lease renewal
-	go vault.PeriodicallyRenewSecrets(ctx, authToken, databaseCredentialsLease, database.Reconnect)
+	// start the secret renewal goroutine & join it on exit
```

Comment:
What did you mean by "join it on exit" here?

Reply:
"join" is more of a thread-related term (used in most thread-heavy languages, e.g. [c++](https://www.cplusplus.com/reference/thread/thread/join/)). It's analogous to joining two wool thread together 😄 . I can change it to "wait for it on exit" if that makes it clearer.

Reply:
Ohhh okay, I have not done C++ development before so that might be why. If you think it's commonly understood by Go developers feel free to keep it, otherwise I think "wait for it to exit" would be better. (with "to" rather than "on", since we're waiting for it to exit, not waiting after it exits)

Reply:
I changed to "wait for it to finish on exit"